test.sh

```diff
@@ -0,0 +1,208 @@
+  window=$initial_window
+  pane=$initial_pane
+  
+  # Determine number of machines avaiable for use when using stu
+  if [[ "$hostname" == *"stu.cs.jmu.edu"* ]]
+    then
+      let "machine_cnt=$(wc -l < configs/stu_machines.txt)"
+  fi
+  
+  tmux has-session -t bdwstest
+
+  if [ $? != 0 ]
+    then
+      # Tmux window with worker_cnt worker panes, a supervisor pane, and a client pane
+      tmux new-session -s bdwstest -n "bdws-test" -d
+    
+      ### SUPERVISOR PANE (bdwstest:0.0) ###
+      #tmux split-window -h -t bdwstest:$window.$pane
+      tmux send-keys -t bdwstest:$window.$pane "cd $PROJECT_DIRECTORY" Enter
+      tmux send-keys -t bdwstest:$window.$pane "cd cmd/supervisor" Enter
+      tmux send-keys -t bdwstest:$window.$pane "go build" Enter
+      tmux send-keys -t bdwstest:$window.$pane "./supervisor :$sup_port" Enter
+      tmux send-keys -t bdwstest:$window.$pane "sleep 1" Enter
+      
+      ### CLIENT PANE (bdwstest:0.1) ###
+      tmux split-window -h -t bdwstest:$window.$pane
+      let "pane++"
+      let "worker_port++"
+      # Check stu lab machine, when hostname contains stu
+      if [[ "$hostname" == *"stu.cs.jmu.edu"* ]]
+        then
+	  let "machine_num=machine_num%$machine_cnt+1"
+	  lab_machine=$(sed -n ${machine_num}p configs/stu_machines.txt)
+          tmux send-keys -t bdwstest:$window.$pane "ssh -o StrictHostKeyChecking=no $lab_machine" Enter
+      fi
+      tmux send-keys -t bdwstest:$window.$pane "cd $PROJECT_DIRECTORY" Enter
+      tmux send-keys -t bdwstest:$window.$pane "cd cmd/client" Enter
+      tmux send-keys -t bdwstest:$window.$pane "go build" Enter
+      tmux send-keys -t bdwstest:$window.$pane "sleep 1" Enter
+      
+      let "pane_count=$worker_cnt+1"
+
+      ### WORKER PANES (bdwstest:window.pane) ###
+      for i in $(seq 2 $pane_count)
+        do
+    	  # Create Worker i
+	  let "pane=i%$max_pane_cnt"
+	  # If current pane is final pane of window, don't split window. Otherwise split the window
+	  if [ $pane -eq 0 ]
+	    then
+	      tmux select-pane -t bdwstest:$window.$pane
+	  else
+	      let "pane_tmp=$pane-1"
+              tmux split-window -v -t bdwstest:$window.$pane_tmp
+	  fi
+	  # Worker i will corespond to machine (5000 + i+1), this is offset by 1 with supervisor starting on 5001
+	  let "worker_port++"
+	  # Check stu lab machine, when hostname contains stu
+	  if [[ "$hostname" == *"stu.cs.jmu.edu"* ]]
+            then
+	      let "machine_num=machine_num%$machine_cnt+1"
+	      lab_machine=$(sed -n ${machine_num}p configs/stu_machines.txt)
+              tmux send-keys -t bdwstest:$window.$pane "ssh -o StrictHostKeyChecking=no $lab_machine" Enter
+	  fi
+
+	  # Startup Worker i
+	  tmux send-keys -t bdwstest:$window.$pane "cd $PROJECT_DIRECTORY" Enter
+          tmux send-keys -t bdwstest:$window.$pane "cd cmd/worker" Enter
+          tmux send-keys -t bdwstest:$window.$pane "go build" Enter
+          tmux send-keys -t bdwstest:$window.$pane "sleep 1" Enter
```

Comment:
It may be a good idea to have workers sleep 2 seconds instead of 1 before trying to connect. I ran into a connection error while testing and a longer wait would solve this.